signed/sign.go

```diff
@@ -7,11 +7,19 @@ import (
 )
 
 func Sign(s *data.Signed, k *data.Key) {
+	id := k.ID()
+	signatures := make([]data.Signature, 0, len(s.Signatures))
```

Comment:
I assumed `append` returns a new slice rather than modifying the given slice? If that is not the case, why do we do `slice = append(slice, ...)` rather than just `append(slice, ...)`?